src/decode.rs

```diff
@@ -4587,172 +4587,42 @@ unsafe fn decode_b(
             }
         } else {
-            if (*f.frame_hdr).frame_type as libc::c_uint
-                & 1 as libc::c_int as libc::c_uint != 0
-                && b.c2rust_unnamed.c2rust_unnamed_0.comp_type as libc::c_int
-                    == COMP_INTER_NONE as libc::c_int
-                && b.c2rust_unnamed.c2rust_unnamed_0.motion_mode as libc::c_int
-                    == MM_WARP as libc::c_int
-            {
-                if b
-                    .c2rust_unnamed
-                    .c2rust_unnamed_0
-                    .c2rust_unnamed
-                    .c2rust_unnamed_0
-                    .matrix[0 as libc::c_int as usize] as libc::c_int
-                    == -(32767 as libc::c_int) - 1 as libc::c_int
-                {
+            if is_inter_or_switch(&*f.frame_hdr) {
+                if b.matrix()[0] == i16::MIN {
                     t.warpmv.type_0 = DAV1D_WM_TYPE_IDENTITY;
                 } else {
                     t.warpmv.type_0 = DAV1D_WM_TYPE_AFFINE;
-                    (*t)
-                        .warpmv
-                        .matrix[2 as libc::c_int
-                        as usize] = b
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .matrix[0 as libc::c_int as usize] as libc::c_int
-                        + 0x10000 as libc::c_int;
-                    (*t)
-                        .warpmv
-                        .matrix[3 as libc::c_int
-                        as usize] = b
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .matrix[1 as libc::c_int as usize] as int32_t;
-                    (*t)
-                        .warpmv
-                        .matrix[4 as libc::c_int
-                        as usize] = b
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .matrix[2 as libc::c_int as usize] as int32_t;
-                    (*t)
-                        .warpmv
-                        .matrix[5 as libc::c_int
-                        as usize] = b
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .matrix[3 as libc::c_int as usize] as libc::c_int
-                        + 0x10000 as libc::c_int;
+                    t.warpmv.matrix[2] = b.matrix()[0] as int32_t + 0x10000;
+                    t.warpmv.matrix[3] = b.matrix()[1] as int32_t;
+                    t.warpmv.matrix[4] = b.matrix()[2] as int32_t;
+                    t.warpmv.matrix[5] = b.matrix()[3] as int32_t + 0x10000;
```

Comment:
This one used to be `as c_int`, not `as int32_t`, too.

Reply:
Resolved here: https://github.com/memorysafety/rav1d/pull/132#discussion_r1169106458.